### PR TITLE
Add sorting and filtering support for nested objects

### PR DESCRIPTION
**Pull request by JWThewes**
Due to sorting and filtering not working for nested objects I added this functionality. Tests have been enhanced to validate this functionality.
Extracted retrieval of nested object values into data.ts

### CODE REVIEW

File: src/__stories__/00-Uncontrolled.test.tsx [hunk 1]
```diff
@@ -1095,10 +1095,14 @@ describe('composed table rows', () => {
 
 describe('Nested object', () => {
   test('nested object table renders', () => {
-    const { getByRole } = render(<NestedObjectTable />);
+    const { getByRole,getByText } = render(<NestedObjectTable />);
```
Comment (imballinst) [inline src/__stories__/00-Uncontrolled.test.tsx:1098]: **The most minor thing in the world:** could we prettify this file? There seems to be no space between `,` and `getByText`.

File: src/helpers/data.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {
   TableColumnType,
   TableRowType
 } from './types';
+import { el } from 'date-fns/locale';
```
Comment (imballinst) [inline src/helpers/data.ts:7]: I assume this is accidentally imported? 👀

File: src/helpers/data.ts [hunk 3]
```diff
@@ -76,7 +86,7 @@ export function filterData<TTableRowType extends TableRowType>(
       const header = headers[key];
 
       if (header.isFilterable) {
-        let columnValue = element[header.prop];
+        let columnValue = extractValueFromObject(header.prop,element);
```
Comment (imballinst) [inline src/helpers/data.ts:89]: (similar like the first comment, let's prettify this file so there's a space between `,` and `element`.

File: src/helpers/data.ts [hunk 4]
```diff
@@ -53,6 +54,15 @@ export function sortData<TTableRowType extends TableRowType>(
   return sortedData;
 }
 
+export function extractValueFromObject(prop: string, rowData:any) {
```
Comment (imballinst) [inline src/helpers/data.ts:57]: Cool stuff! Thanks for creating this utility function.

**Review by imballinst — COMMENTED** [review]
Thanks @JWThewes for working on this! I have only 2 overall feedback:

1. First one is about formatting (we can just prettify the changed files), and
2. Could you help add a changeset? There's a section [in the main README.md](https://github.com/imballinst/react-bs-datatable?tab=readme-ov-file#contributing) about it. Let me know if you have questions.

**Comment by JWThewes** [comment]
I made the changes according to you comments.

**Review by imballinst — APPROVED** [review]
Perfect. Thanks @JWThewes for working on this! Since we're using automation for the release, I suppose we can release it soon (maybe in <1 hour, hopefully).

**Comment by imballinst** [comment]
Latest version should be now available in npm: https://www.npmjs.com/package/react-bs-datatable/v/3.15.0. Thanks again @JWThewes for the PR!

**Pull request merged**